src/entities/trade.ts

```diff
@@ -114,10 +166,20 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
    * Returns the percent difference between the route's mid price and the price impact
    */
   public get priceImpact(): Percent {
-    return (
-      this._priceImpact ??
-      (this._priceImpact = computePriceImpact(this.route.midPrice, this.inputAmount, this.outputAmount))
-    )
+    if (this._priceImpact) {
+      return this._priceImpact
+    }
+
+    let spotOutputAmount = CurrencyAmount.fromRawAmount(this.outputAmount.currency, 0)
+    for (const { route, inputAmount } of this.routes) {
+      const midPrice = route.midPrice
+      spotOutputAmount = spotOutputAmount.add(midPrice.quote(inputAmount))
```

Comment:
mmmm is this correct... i don't really have an intuition about the meaning of price impact across routes

Reply:
I think so, and I think @NoahZinsmeister agrees?

Reply:
this is what i suggested, it seems consistent with how we were calculating it in the single-route case, and works out to a weighted average price impact across routes, where the weights are the quote amounts and the values are the individual price impact